build-system/tasks/presubmit-checks.js

```diff
@@ -1439,36 +1435,26 @@ function isMissingTerms(file) {
 /**
  * Check a file for all the required terms and
  * any forbidden terms and log any errors found.
- * @return {!Promise}
  */
-function presubmit() {
+async function presubmit() {
   let forbiddenFound = false;
   let missingRequirements = false;
-  return gulp
-    .src(srcGlobs)
-    .pipe(
-      through2.obj(function (file, enc, cb) {
-        forbiddenFound = hasAnyTerms(file) || forbiddenFound;
-        missingRequirements = isMissingTerms(file) || missingRequirements;
-        cb();
-      })
-    )
-    .on('end', function () {
-      if (forbiddenFound) {
-        log(blue('Please remove these usages or consult with the AMP team.'));
-      }
-      if (missingRequirements) {
-        log(
-          blue(
-            'Adding these terms (e.g. by adding a required LICENSE ' +
-              'to the file)'
-          )
-        );
-      }
-      if (forbiddenFound || missingRequirements) {
-        process.exitCode = 1;
-      }
-    });
+  const srcFiles = globby.sync(srcGlobs);
+  for (const srcFile of srcFiles) {
+    forbiddenFound = hasAnyTerms(srcFile) || forbiddenFound;
+    missingRequirements = isMissingTerms(srcFile) || missingRequirements;
```

Comment:
```javascript
const srcFiles = globby.sync(srcGlobs);
const forbiddenFound = srcFiles.any(hasAnyTerms);
const missingRequirements = srcFiles.any(isMissingTerms);
```

Reply:
This is a neat idea, but will exit as soon as the first error is detected, and skip checking the remaining files. Also, it's `Array.some`, not `Array.any` :)

Reply:
Looks like my Python is showing (any/some)!

I had assumed based on the predicate naming that these functions were idempotent. If they have side-effects, renaming to something like `checkForbiddenTerms` etc. might be clearer. Fine with punting on this though